tasks/hyrax-metadata-updates/tests/test-integration-index.js

```diff
@@ -312,3 +313,108 @@ test('Test generate path from ECHO-10 throws exception with broken config', asyn
   });
 });
+
+test.serial('Test updating ECHO10 metadata file in S3 when formats are present and correct', async (t) => {
+  // Set up mock Validation call to CMR
+  nock('https://cmr.earthdata.nasa.gov').post('/ingest/providers/GES_DISC/validate/granule/MOD11A1.A2017200.h19v04.006.2017201090724.cmr.xml')
+    .reply(200);
+  await setupS3(t, false, 'tests/data/echo10inWithFormat.xml');
+  const config = cloneDeep(event.config);
+  config.hyrax = {
+    formats: [
+      'HDF-5'
+    ]
+  };
+  const e = {
+    config: config,
+    input: t.context.payload.input
+  };
+
+  await hyraxMetadataUpdate(e);
+  // Verify the metadata has been updated at the S3 location
+  const metadataFile = t.context.payload.input.granules[0].files.find((f) => f.type === 'metadata');
+  const actual = await getS3Object(`${metadataFile.bucket}/${metadataFile.fileStagingDir}`, metadataFile.name);
+  const expected = fs.readFileSync('tests/data/echo10outWithFormat.xml', 'utf8');
+  t.is(actual.Body.toString(), expected);
+
+  await recursivelyDeleteS3Bucket(t.context.stagingBucket);
+});
+
+test.serial('Test updating ECHO10 metadata file in S3 when formats are present and not matching', async (t) => {
+  await setupS3(t, false, 'tests/data/echo10inWithFormat.xml');
+  const config = cloneDeep(event.config);
+  config.hyrax = {
+    formats: [
+      'NETCDF-4'
+    ]
+  };
+  const e = {
+    config: config,
+    input: t.context.payload.input
+  };
+
+  await hyraxMetadataUpdate(e);
+  // Verify the metadata has not been updated at the S3 location
+  const metadataFile = t.context.payload.input.granules[0].files.find((f) => f.type === 'metadata');
+  const actual = await getS3Object(`${metadataFile.bucket}/${metadataFile.fileStagingDir}`, metadataFile.name);
+  const expected = fs.readFileSync('tests/data/echo10inWithFormat.xml', 'utf8');
+  t.is(actual.Body.toString(), expected);
+
+  await recursivelyDeleteS3Bucket(t.context.stagingBucket);
+});
+
+test.serial('Test updating UMM-G metadata file in S3 when formats are present and correct', async (t) => {
+  // Set up mock Validation call to CMR
+  nock('https://cmr.earthdata.nasa.gov').post('/ingest/providers/GES_DISC/validate/granule/MOD11A1.A2017200.h19v04.006.2017201090724.cmr.json')
+    .reply(200);
+  await setupS3(t, true, 'tests/data/umm-ginWithFormat.json');
+  const config = cloneDeep(event.config);
+  config.hyrax = {
+    formats: [
+      'HDF5'
+    ]
+  };
+  const e = {
+    config: config,
+    input: t.context.payload.input
+  };
+
+  await hyraxMetadataUpdate(e);
+  // Verify the metadata has been updated at the S3 location
+  const metadataFile = t.context.payload.input.granules[0].files.find((f) => f.type === 'metadata');
+  const actual = await getS3Object(`${metadataFile.bucket}/${metadataFile.fileStagingDir}`, metadataFile.name);
+  const expected = fs.readFileSync('tests/data/umm-goutWithFormat.json', 'utf8');
+  // We do this dance because formatting.
```

Comment:
What formatting issue does this workaround?

Reply:
If I compare the JSON objects I get,

Difference:

  - {
  -   Body: Buffer @Uint8Array [ … ],
  -   ContentLength: 1859,
  -   ContentType: 'application/octet-stream',
  -   ETag: '"e7038328e1354ad4fd17cc461595926c"',
  -   LastModified: Date 2020-03-18 21:12:02 UTC {},
  -   Metadata: {},
  - }
  + `{␊
  +     "RelatedUrls": [␊
  +         {␊
  +             "URL": "https://hydro1.gesdisc.eosdis.nasa.gov/data/GLDAS/GLDAS_CLSM025_D.2.0/2014/12/GLDAS_CLSM025_D.A20141230.020.nc4",␊
  +             "Type": "GET DATA"␊
  +         },␊
  +         {␊
  +             "URL": "https://opendap.earthdata.nasa.gov/providers/GES_DISC/collections/GLDAS%20Catchment%20Land%20Surface%20Model%20L4%20daily%200.25%20x%200.25%20degree%20V2.0%20(GLDAS_CLSM025_D)%20at%20GES%20DISC/granules/GLDAS_CLSM025_D.2.0:GLDAS_CLSM025_D.A20141230.020.nc4",␊
  +             "Type": "GET DATA",␊
  +             "Subtype": "OPENDAP DATA",␊
  +             "Description": "OPeNDAP request URL"␊
  +         }␊
  +     ],␊
  +     "SpatialExtent": {␊
  +         "HorizontalSpatialDomain": {␊
  +             "Geometry": {␊
  +                 "BoundingRectangles": [␊
  +                     {␊
  +                         "WestBoundingCoordinate": -180.0,␊
  +                         "EastBoundingCoordinate": 180.0,␊
  +                         "NorthBoundingCoordinate": 90.0,␊
  +                         "SouthBoundingCoordinate": -60.0␊
  +                     }␊
  +                 ]␊
  +             }␊
  +         }␊
  +     },␊
  +     "ProviderDates": [␊
  +         {␊
  +             "Date": "2018-02-09T11:01:38.000Z",␊
  +             "Type": "Insert"␊
  +         },␊
  +         {␊
  +             "Date": "2018-02-09T11:01:38.000Z",␊
  +             "Type": "Update"␊
  +         }␊
  +     ],␊
  +     "CollectionReference": {␊
  +         "ShortName": "GLDAS_CLSM025_D",␊
  +         "Version": "2.0"␊
  +     },␊
  +     "DataGranule": {␊
  +         "DayNightFlag": "Unspecified",␊
  +         "Identifiers": [␊
  +             {␊
  +                 "Identifier": "GLDAS_CLSM025_D.A20141230.020.nc4",␊
  +                 "IdentifierType": "ProducerGranuleId"␊
  +             }␊
  +         ],␊
  +         "ProductionDateTime": "2018-02-09T11:01:38.000Z",␊
  +         "ArchiveAndDistributionInformation": [␊
  +             {␊
  +                 "Name": "Not provided",␊
  +                 "Size": 24.5028533935547,␊
  +                 "SizeUnit": "MB",␊
  +                 "Format": "HDF5"␊
  +             }␊
  +         ]␊
  +     },␊
  +     "TemporalExtent": {␊
  +         "RangeDateTime": {␊
  +             "BeginningDateTime": "2014-12-30T00:00:00.000Z",␊
  +             "EndingDateTime": "2014-12-30T23:59:59.000Z"␊
  +         }␊
  +     },␊
  +     "GranuleUR": "GLDAS_CLSM025_D.2.0:GLDAS_CLSM025_D.A20141230.020.nc4"␊
  + }`